src/server/auth/server/testing/admin_test.go

```diff
@@ -262,116 +436,89 @@ func TestAdminFixBrokenRepo(t *testing.T) {
-	if testing.Short() {
-		t.Skip("Skipping integration tests in short mode")
-	}
-	deleteAll(t)
-	defer deleteAll(t)
-	alice := tu.UniqueString("alice")
-	aliceClient, adminClient := getPachClient(t, alice), getPachClient(t, admin)
-
-	// Check that the initial set of admins is just "admin"
-	resp, err := adminClient.GetAdmins(adminClient.Ctx(), &auth.GetAdminsRequest{})
-	require.NoError(t, err)
-	require.ElementsEqual(t, []string{admin}, resp.Admins)
-
-	// admin cannot remove themselves from the list of cluster admins (otherwise
-	// there would be no admins)
+	// admin cannot remove themselves from the list of super admins (otherwise
+	// there would be no admins), alice only has fs admin
 	_, err = adminClient.ModifyAdmins(adminClient.Ctx(),
-		&auth.ModifyAdminsRequest{Remove: []string{admin}})
+		&auth.ModifyAdminsRequest{Principal: admin})
 	require.YesError(t, err)
 	require.NoError(t, backoff.Retry(func() error {
 		resp, err = adminClient.GetAdmins(adminClient.Ctx(), &auth.GetAdminsRequest{})
 		require.NoError(t, err)
-		return require.ElementsEqualOrErr([]string{admin}, resp.Admins)
+		return require.EqualOrErr(admins(admin)(gh(alice)), resp.Admins)
 	}, backoff.NewTestingBackOff()))
 
 	// admin can make alice a cluster administrator
 	_, err = adminClient.ModifyAdmins(adminClient.Ctx(),
 		&auth.ModifyAdminsRequest{
-			Add: []string{alice},
+			Principal: gh(alice),
+			Roles:     superAdminRole(),
 		})
 	require.NoError(t, err)
 	require.NoError(t, backoff.Retry(func() error {
 		resp, err = adminClient.GetAdmins(adminClient.Ctx(), &auth.GetAdminsRequest{})
 		require.NoError(t, err)
-		return require.ElementsEqualOrErr([]string{admin, gh(alice)}, resp.Admins)
+		return require.EqualOrErr(admins(admin, gh(alice))(), resp.Admins)
 	}, backoff.NewTestingBackOff()))
 
 	// Now admin can remove themselves as a cluster admin
 	_, err = adminClient.ModifyAdmins(adminClient.Ctx(),
-		&auth.ModifyAdminsRequest{Remove: []string{admin}})
+		&auth.ModifyAdminsRequest{Principal: admin})
 	require.NoError(t, err)
 	require.NoError(t, backoff.Retry(func() error {
 		resp, err = adminClient.GetAdmins(adminClient.Ctx(), &auth.GetAdminsRequest{})
 		require.NoError(t, err)
-		return require.ElementsEqualOrErr([]string{gh(alice)}, resp.Admins)
+		return require.EqualOrErr(admins(gh(alice))(), resp.Admins)
 	}, backoff.NewTestingBackOff()))
 
 	// now alice is the only admin, and she cannot remove herself as a cluster
 	// administrator
 	_, err = aliceClient.ModifyAdmins(aliceClient.Ctx(),
-		&auth.ModifyAdminsRequest{Remove: []string{alice}})
+		&auth.ModifyAdminsRequest{Principal: alice})
 	require.YesError(t, err)
 	require.NoError(t, backoff.Retry(func() error {
 		resp, err = aliceClient.GetAdmins(aliceClient.Ctx(), &auth.GetAdminsRequest{})
 		require.NoError(t, err)
-		return require.ElementsEqualOrErr([]string{gh(alice)}, resp.Admins)
-	}, backoff.NewTestingBackOff()))
-
-	// alice *can* swap herself and "admin"
```

Comment:
Just noting for posterity, because `modify-admins` is now `modify-admin`, swapping one admin for another isn't something that can be done transactionally, so there's nothing to test.